src/Spotify/SpotifyAPI.cpp

```diff
@@ -379,65 +270,147 @@ TResultOpt SpotifyAPI::play(std::string const &accessToken,
     body["uris"] = uris;
   }
 
-  client->SetTimeout(cRequestTimeout);
-  auto response =
-      client->put("/v1/me/player/play" + queryStream.str(), body.dump());
+  auto responseRet = spotifyCall(accessToken,
+                                 "/v1/me/player/play",
+                                 HttpPut,
+                                 queryStream.str(),
+                                 body.dump());
+  if (auto value = std::get_if<Error>(&responseRet)) {
+    return *value;
+  }
+  auto response = std::get<RestClient::Response>(responseRet);
 
   if (response.code == cNoContent) {
     return std::nullopt;
   }
 
-  nlohmann::json errJson;
-  try {
-    errJson = nlohmann::json::parse(response.body);
-
-    // check for error object
-    if (errJson.find("error") != errJson.end()) {
-      SpotifyError spotifyError(errJson["error"]);
-      return errorParser(spotifyError);
-    }
-
-    // if we reach here or the exception gets thrown spotify sent an unexpected
-    // message
-  } catch (...) {
-    // parse exception
-    return Error(ErrorCode::SpotifyParseError,
-                 "Received json couldn't be parsed");
+  auto ret = parseSpotifyCall<SpotifyDummy>(response);
+  if (auto error = std::get_if<Error>(&ret)) {
+    return *error;
   }
-  // if we reach here spotify sent an unexpected message
-  return Error(ErrorCode::SpotifyAPIError,
-               "Spotify sent an unexpected message");
+
+  // we should never reach here because this endpoint only sends Error or
+  // noContent
+  LOG(ERROR) << "SpotifyAPI.setVolume: Fatal Error" << std::endl;
+  return Error(ErrorCode::SpotifyAPIError, "SpotifyAPI.play: Fatal Error");
 }
 
 TResult<Track> SpotifyAPI::getTrack(std::string const &accessToken,
                                     std::string const &spotifyID,
                                     const std::string &market) {
   (void)market;
-  auto client = std::make_unique<RestClient::Connection>(cSpotifyAPIUrl);
+
+  auto responseRet =
+      spotifyCall(accessToken, "/v1/tracks/" + spotifyID, HttpGet);
+  if (auto value = std::get_if<Error>(&responseRet)) {
+    return *value;
+  }
+  auto response = std::get<RestClient::Response>(responseRet);
+
+  if (response.code == cNoContent) {
```

Comment:
shouldnt `cNoContent` be an error when querying for a track? from the JukeBoxs point of view thats an unexpected result

Reply:
oh yeah you're right, will change it